microsoft/atlas-design
# VSCode Extension for Class Autocompletion, Swatches, documentation links

Task: hackathon-2022

Link: preview-465

Creates a VSCode extension in /extension.

Updates /css to build a json data structure that can be used to populate suggestions.

Updates /site to adorn pages with `classType` and `classPrefix`. These are used to provide extra information pertaining to suggestions.

Updates publishing pipeline to publish new extension versions when atlas-css is published.

## Testing

1. Pull down branch. `gh pr checkout 465`
2. `npm i`
3. `npm run prepare:extension`, this task ensures the several generated elements are in place prior to debugging the extension.
4. Press "F5" select "Run extension"
5. A new VSCode window will open with the [Extension Host] label. Open a .ts or .html file. Start typing an atlas class name. You should see a list populating intellisense. `color-` etc.
6. You'll be able to click the righthand side of an item (on a chevron) and see extra info, including color swatches and padding/margin values.
